analyses/oncoprint-landscape/01-plot-oncoprint.R

```diff
@@ -0,0 +1,241 @@
+
+# Declare command line options
+option_list <- list(
+  optparse::make_option(
+    c("-m", "--maf_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to MAF file that contains snv information",
+  ),
+  optparse::make_option(
+    c("-c", "--cnv_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to SEG file that contains cnv information"
+  ),
+  optparse::make_option(
+    c("-f", "--fusion_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to file that contains fusion information"
+  ),
+  optparse::make_option(
+    c("-g", "--goi_list"),
+    type = "character",
+    default = NULL,
+    help = "file path to file that contains list of genes to include on
+            oncoprint"
+  ),
+  optparse::make_option(
+    c("-p", "--png_name"),
+    type = "character",
+    default = NULL,
+    help = "oncoprint output png file name"
+  )
+)
+
+# Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+maf <- opt$maf_file
+
+# Define cnv_file object here as it still needs to be defined for the `read.maf`
+# function, even if it is NULL
+cnv_file <- opt$cnv_file
+
+#### Read in data --------------------------------------------------------------
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv"))
+
+# Rename for maftools function
+metadata <- metadata %>%
+  dplyr::rename("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")
+
+# Read in MAF file
+maf_df <- data.table::fread(maf, stringsAsFactors = FALSE)
+
+# Read in cnv file
+if (!is.null(opt$cnv_file)) {
+  cnv_file <- data.table::fread(opt$cnv_file, stringsAsFactors = FALSE)
+  # TODO: Filter and set up `cnv_file` to be in the column format -
+  # "Hugo_Symbol, Tumor_Sample_Barcode, Variant_Classification" as required by
+  # the `read.maf function`
+}
+
+# Read in fusion file
+if (!is.null(opt$fusion_file)) {
+  fusion_file <- data.table::fread(opt$fusion_file, stringsAsFactors = FALSE)
+
+  #### Incorporate Fusion Data -------------------------------------------------
+
+  # Separate fusion gene partners and add variant classification and center
+  fus_sep <- fusion_file %>%
+    tidyr::separate(FusionName, c("5'-gene", "3'-gene"), sep = "--")
+
+  # Determine which samples have multi-fused fusions
+  multi <- fus_sep[, c("Sample", "5'-gene", "3'-gene")]
+  multi$ID <- paste0(multi$Sample, ";", multi$`5'-gene`)
+  reformat_multi <- multi %>%
+    dplyr::group_by(ID) %>%
+    dplyr::summarise(Sample = dplyr::n()) %>%
+    as.data.frame()
+
+  reformat_multi$Variant_Classification <-
+    ifelse(reformat_multi$Sample == 1, "Fusion", "Multi_Hit_Fusion")
+  reformat_multi <- reformat_multi %>%
+    tidyr::separate(ID, c("Tumor_Sample_Barcode", "Hugo_Symbol"), sep = ";")
+  reformat_multi$Sample <- NULL
+  reformat_multi$Variant_Type <- "OTHER"
```

Comment:
I think this section would benefit from using more R-friendly column names and it could be simplified by using additional dplyr functionality:

```R
fus_sep <- fusion_file %>%
  # can add a note here about 5' and 3'
  tidyr::separate(FusionName, c("Gene1", "Gene2"), sep = "--") %>%  
  dplyr::select(Sample, Gene1, Gene2)

reformat_fusion <- fus_sep %>% 
  # here we want to tally how many times the 5' gene shows up as a fusion hit 
  # in a sample
  dplyr::group_by(Sample, Gene1) %>%
  dplyr::tally() %>%
  # if the sample-5' gene pair shows up more than once, call it a multi hit 
  # fusion
  dplyr::mutate(Variant_Classification = 
                  dplyr::if_else(n == 1, "Fusion", "Multi_Hit_Fusion"),
                # required column for joining with MAF
                Variant_Type = "OTHER") %>%
  # correct format for joining with MAF
  dplyr::rename(Tumor_Sample_Barcode = Sample, Hugo_Symbol = Gene1) %>%
  dplyr::select(-n)
```

Should be equivalent. Then you're ready to bind `reformat_fusion`.